libsolidity/parsing/Scanner.h

```diff
@@ -162,6 +162,7 @@ class Scanner
 	/// Do only use in error cases, they are quite expensive.
 	std::string lineAtPosition(int _position) const { return m_source.lineAtPosition(_position); }
 	std::tuple<int, int> translatePositionToLineColumn(int _position) const { return m_source.translatePositionToLineColumn(_position); }
+	std::string printSourceLocation(SourceLocation const& _location) const { return m_source.source().substr(_location.start, _location.end - _location.start); }
```

Comment:
Do we need to care about the source location (start and/or end) possibly being ``-1`` here? If I recall correctly, this can happen under some circumstances - not sure whether it applies here, though.